services/outside_network.c

```diff
@@ -1914,11 +1940,20 @@ randomize_and_send_udp(struct pending* pend, sldns_buffer* packet, int timeout)
 	comm_timer_set(pend->timer, &tv);
 
 #ifdef USE_DNSTAP
+	/*
+	 * sending src (local service)/dst (upstream) addresses over DNSTAP
+	 * TODO: right now there are no chances to get the src (local service) addr. So we will pass 0.0.0.0 (::)
+	 * to argument for dt_msg_send_outside_query()/dt_msg_send_outside_response() calls.
+	 * For the both UDP and TCP.
```

Comment:
Yes, but also no, in some cases the user could has configured IP-addresses, but when unbound starts with defaults, or with config of the wildcard, the 0.0.0.0 is in the addr pointer that is passed here, and it is not the address of the interface that the OS chooses after unbound uses that wildcard.

Reply:
Then maybe:
```
There are no chances to get the src (local service) addr if unbound is not configured with specific outgoing IP-addresses. So we will pass 0.0.0.0 (::) to argument for dt_msg_send_outside_query()/dt_msg_send_outside_response() calls.
```
And also remove the "For the both UDP and TCP." line since here we are only doing UDP?

Reply:
Fixed it like you suggested in https://github.com/NLnetLabs/unbound/pull/367/commits/04b4f515a7b3de1dc02d78990f7a0c3d1d6d84aa    I think that is a better comment.  It is also true for TCP, but we need not spam it everywhere we call this function.